lib/senkyoshi/collection.rb

```diff
@@ -16,6 +16,24 @@ def detect_xid(xid)
       end
     end
 
+    def find_by_id(id)
+      @resources.detect do |item|
```

Comment:
I'm tempted to make this a one liner `@resources.detect { |item| item.respond_to?(:id) && item.id == id }` but that is up to you.